Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAutomationWorker.py

```diff
@@ -16,212 +15,133 @@
 import imp
 protocol = imp.load_source('protocol', '../protocol.py')
 nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+try:
+    serializerfactory = imp.load_source('serializerfactory', '../../modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/worker/serializerfactory.py')
+except:
+    # this is the path when running tests
+    serializerfactory = imp.load_source('serializerfactory', '../../nxOMSAutomationWorker/automationworker/worker/serializerfactory.py')
 LG = nxDSCLog.DSCLog
 
+def Set_Marshall(ResourceSettings):
+    workspace_id, azure_dns_agent_svc_zone, updates_enabled, diy_enabled = read_settings_from_mof_json(ResourceSettings)
+    if not is_oms_primary_workspace(workspace_id):
+        # not primar workspace
+        # return unconditional [0] for a NOOP on non-primary workspace
+        log(DEBUG, "Set_Marshall skipped: non primary workspace. Set marshall returned [0]")
+        return [0]
 
-def init_locals(WorkspaceId, AzureDnsAgentSvcZone):
-    if WorkspaceId is None:
-        WorkspaceId = ''
-    if AzureDnsAgentSvcZone is None:
-        AzureDnsAgentSvcZone = ''
-    return WorkspaceId.encode('ascii', 'ignore'), AzureDnsAgentSvcZone.encode('ascii', 'ignore')
+    try:
+        # compatibility from 1.3 remove state.conf file
+        if os.path.isfile(STATE_CONF_FILE_PATH):
+            os.remove(STATE_CONF_FILE_PATH)
+
+        # Kill worker managers that might already be running
+        kill_worker_manager(workspace_id)
```

Comment:
Minor: consider adding a few more traces after important calls this one. Once we go public the only way we will have to debug customer's machine if the worker isn't starting will be the omiconfig logs.